### PR TITLE
Bluetooth: host: Fix non-connectable advertiser configured as scannable

### LINKED ISSUE #37462: Bluetooth: Advertising becomes scannable even if BT_LE_ADV_OPT_FORCE_NAME_IN_AD is set
**Describe the bug**
When creating an extended advertiser or just using the `bt_le_adv_start` with the `BT_LE_ADV_OPT_USE_NAME` option, it implicitly makes the advertisements scannable, as it would put the name in the scan response data. 

However, we introduced another option, `BT_LE_ADV_OPT_FORCE_NAME_IN_AD` (https://github.com/zephyrproject-rtos/zephyr/pull/34332), which forces the name to be in the advertising data, but the advertisement will still become scannable, even if the upper layer didn't request it to be. 

**To Reproduce**
N/A

**Expected behavior**
If `BT_LE_ADV_OPT_FORCE_NAME_IN_AD` is set with `BT_LE_ADV_OPT_USE_NAME`, `BT_LE_ADV_OPT_SCANNABLE` isn't set and no scan response data was provided, then the advertisement should not be scannable.

This can be fixed by changing how we determine whether it should be scannable from doing 
```
scannable = param->options & BT_LE_ADV_OPT_USE_NAME;
```
to
```
scannable = (param->options & BT_LE_ADV_OPT_USE_NAME != 0) && (param->options & BT_LE_ADV_OPT_FORCE_NAME_IN_AD == 0);
```

**Impact**
Minor, will just making it scannable without any scan response data. 

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Commit SHA or Version used: 1c0ed946018e4f1ee4b766d5db811d097f44f2ba

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by joerchan**
Fix non-connectable advertiser configured as ADV_SCAN_IND when configured
by application to have the device name appear in the advertising data
instead of the scan response data.

Disallow creating an extended scannable advertiser with the device name
configured to appear in the advertising data.
This would fail in either the call to bt_le_ext_adv_start or
bt_le_ext_adv_set_data when the host would try to set advertising data in the
controller.
Instead this now fails in the bt_le_ext_adv_create call.

Document the behavior of the include name option when this is changed by
updating the advertising parameters of the advertising set.
In this case we cannot update the advertising data since the host does not have
a copy to modify. The application will have to do update with its current
advertising data.

Allow the application to configure the advertiser as scannable when it does not
provide scan data in the call to bt_le_adv_start. This makes it possible for the
application to later add scan response data in the bt_le_adv_update_data call.
This aligns the legacy code path with the extended code path which already had
this behavior.

This also stops a directed connectable advertiser from being marked internally
in the host as scannable. This appears to not have been causing any issues.

Fixes: #37462

### CODE REVIEW

File: subsys/bluetooth/host/adv.c [hunk 1]
```diff
@@ -459,8 +495,8 @@ static bool ad_is_limited(const struct bt_data *ad, size_t ad_len)
 static int le_adv_update(struct bt_le_ext_adv *adv,
 			 const struct bt_data *ad, size_t ad_len,
 			 const struct bt_data *sd, size_t sd_len,
-			 bool ext_adv, bool scannable, bool use_name,
-			 bool force_name_in_ad)
+			 bool ext_adv, bool scannable,
+			 enum adv_name_type use_name)
```
Comment (Thalley) [inline subsys/bluetooth/host/adv.c:499]: `use_name` sounds like a boolean value. May I suggestion something like `name_type` instead?

File: subsys/bluetooth/host/adv.c [hunk 2]
```diff
@@ -786,6 +822,7 @@ static int le_ext_adv_param_set(struct bt_le_ext_adv *adv,
 	bool dir_adv = param->peer != NULL, scannable;
 	struct net_buf *buf, *rsp;
 	int err;
+	enum adv_name_type adv_name;
```
Comment (Thalley) [inline subsys/bluetooth/host/adv.c:825]: Isn't `name_type` a better variable name than `adv_name`? 

Alternatively rename the enum to `bt_adv_name_type` and use the full `adv_name_type` for the variable.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/adv.c (hunk 1 above)
Comment (joerchan) [reply subsys/bluetooth/host/adv.c]: Thanks, that's better.

File: subsys/bluetooth/host/adv.c (hunk 2 above)
Comment (joerchan) [reply subsys/bluetooth/host/adv.c]: Renamed to name_type.

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**